code/modules/projectiles/projectile/energy.dm

```diff
@@ -93,6 +92,58 @@
 	tesla_zap(src, zap_range, power, zap_flags)
 	qdel(src)
 
+/obj/item/projectile/energy/bsg
+	name = "orb of pure bluespace energy"
+	icon_state = "bluespace"
+	impact_effect_type = /obj/effect/temp_visual/bsg_kaboom
+	damage = 60
+	damage_type = BURN
+	range = 9
+	weaken = 1 //This is going to knock you off your feet
+	eyeblur = 5
+	speed = 2
+	alwayslog = TRUE
+
+/obj/item/ammo_casing/energy/bsg/ready_proj(atom/target, mob/living/user, quiet, zone_override = "")
+	..()
+	var/obj/item/projectile/energy/bsg/P = BB
+	spawn(1)
+		P.chain = P.Beam(user, icon_state = "sm_arc_supercharged", icon = 'icons/effects/beam.dmi', time = 1000, maxdistance = 30)
+
+/obj/item/projectile/energy/bsg/on_hit(atom/target)
+	. = ..()
+	kaboom()
+	qdel(src)
+
+/obj/item/projectile/energy/bsg/on_range()
+	kaboom()
+	..()
+
+/obj/item/projectile/energy/bsg/proc/kaboom()
+	playsound(src, 'sound/weapons/bsg_explode.ogg', 75, TRUE)
+	for(var/mob/living/M in hearers(7, src)) //No stuning people with thermals through a wall.
+		var/floored = FALSE
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
+			var/obj/item/gun/energy/bsg/N = locate() in H
+			if(N)
+				to_chat(H, "<span class='notice'>[N] deploys an energy shield to project you from the [src]'s explosion.</span>")
+				continue
+		if(prob(min(400 / (1 + get_dist(M, src)), 100)))
```

Comment:
Might be good to use defines here so we know exactly what number does what.

Additionally, I would extract the distance mob-to-projectile into a variable, so it's not computed multiple times.

Reply:
Made it a variable and explained the percentages